core/src/main/scala/org/apache/spark/api/java/JavaRDDLike.scala

```diff
@@ -436,6 +436,12 @@ trait JavaRDDLike[T, This <: JavaRDDLike[T, This]] extends Serializable {
   def first(): T = rdd.first()
 
   /**
+   * @return true if and only if the RDD contains no elements at all. Note that an RDD
+   *         may be empty even when it has at least 1 partition.
+   */
+  def isEmpty(): Boolean = rdd.isEmpty()
```

Comment:
So this is actually a legitimate API break _if_ we think users are themselves extending `JavaRDDLike` trait, because it will add a method to the associated interface. One option is to just do it and ask users not to write code that directly accepts or extends `JavaRDDLike`, and maybe we could document that in the JavaDoc. Another option is just to add this to the concrete implementations in JavaRDD and JavaPairRDD. @JoshRosen, any thoughts one way or the other?